lib/validation.js

```diff
@@ -61,8 +74,19 @@ function validate (handle, params, body, query) {
   return true
 }
 
-function serialize (handle, data) {
-  return handle[outputSchema](data)
+function serializeAccordingToResponseSchema (responseSchema, data, statusCode) {
+  if (responseSchema[statusCode]) {
+    return responseSchema[statusCode](data)
+  }
+  var fallbackStatusCode = (statusCode + '')[0] + 'xx'
+  if (responseSchema[fallbackStatusCode]) {
+    return responseSchema[fallbackStatusCode](data)
+  }
+  return responseSchema.default(data)
+}
+
+function serialize (handle, data, statusCode) {
+  return serializeAccordingToResponseSchema(handle[responseSchema], data, statusCode)
 }
```

Comment:
can you remove this function call completely, and rename `serializeAccordingToResponseSchema`  to `serialize`?